diff --git a/src/stream.c b/src/stream.c
index 89dac69..9bbd8a3 100644
--- a/src/stream.c
+++ b/src/stream.c
@@ -1182,8 +1182,8 @@ uvc_error_t uvc_stream_start(
                                 endpoint_bytes_per_packet - 1) / endpoint_bytes_per_packet;
 
         /* But keep a reasonable limit: Otherwise we start dropping data */
-        if (packets_per_transfer > 32)
-          packets_per_transfer = 32;
+        if (packets_per_transfer > 4)
+          packets_per_transfer = 4;
         
         total_transfer_size = packets_per_transfer * endpoint_bytes_per_packet;
         break;
